### PR TITLE
fix(sql): reject catch-all __getattr__ objects as DB connections

### LINKED ISSUE #10213: marimo notebook hangs when using pytorch-ignite metraics
### Describe the bug

Instantiating a any `ignite.metrics.Metric` instance it just hangs that particular cell executing it. I tried and tested this with `ignite.metrics.Loss` and `ignite.metrics.RunningAverage`. It it just remains hung. It just never terminates, the hourglass keeps spinning...

Image

Further when you try to stop the cell by using the interrupt button, it doesn't get interrupted, here is the log message when clicking that button:

Image

I also checked for the specific codebases of [PyTorch Ignite](https://github.com/pytorch/ignite), more specifically the source codes of `ignite.metrics.Loss` and `ignite.metrics.RunningAverage` but couldn't find anything problematic there. At least I didn't see any problem in their `__init__` method. Please look into this. Or suggest a fix. The issue is same when running locally as well as on _molab_

### Will you submit a PR?

- [ ] Yes

### Environment

<details>

```
OS: Windows 11
Python: 3.14.0 (Miniconda)
marimo: 0.23.14
```

</details>

### Code to reproduce

```python
# In any marimo code cell run the following
import marimo as mo
import torch
from ignite.metrics import Loss, RunningAverage
loss = Loss(torch.nn.CrossEntropyLoss()) # or any other metric practically
```

### PR DESCRIPTION
**Pull request by Light2Dark**
Fixes #10213

## 📝 Summary

Instantiating pytorch-ignite metrics (e.g. `Loss`) hung the cell forever. Those objects implement a catch-all `__getattr__`, so DB-API/ADBC duck-typing treated them as SQL connections; post-execution catalog introspection then never returned.

Add `fabricates_attributes()`: probe an attribute no real connection defines. If it resolves, reject the object in `DBAPIEngine` / `AdbcDBAPIEngine.is_compatible`. Prefer this over `inspect.getattr_static` so connection proxies that forward `__getattr__` still work.

## 📋 Pre-Review Checklist

- [x] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://marimo.io/discord?ref=pr), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [x] Any AI generated code has been reviewed line-by-line by the human PR author, who stands by it.
- [ ] Video or media evidence is provided for any visual changes (optional).

## ✅ Merge Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [x] Documentation has been updated where applicable, including docstrings for API changes.
- [x] Tests have been added for the changes made.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: marimo/_sql/engines/types.py [hunk 1]
```diff
@@ -23,6 +23,23 @@
...
+
+    Used to reject objects (e.g. ignite metrics) that pass every getattr-based
+    duck-type check. Prefer this over `getattr_static` so connection proxies
+    that forward `__getattr__` still work. See https://github.com/marimo-team/marimo/issues/10213.
+    """
+    try:
+        return getattr(var, _MISSING_ATTRIBUTE_PROBE, None) is not None
+    except Exception:
+        # Not a usable connection if __getattr__ raises unexpectedly.
+        return True
```
[the remark this answers is not in the document]
Comment (Light2Dark) [reply marimo/_sql/engines/types.py:47]: Fixed — switched to a unique sentinel so a successful resolution of the probe counts even when the value is `None`, plus a regression test.

**Comment by Light2Dark** [comment]
Addressed Copilot feedback: `fabricates_attributes` now uses a sentinel so `__getattr__` returning `None` is still detected.

**Review by mscolnick — APPROVED** [review]
